warriors.rb

```diff
@@ -0,0 +1,92 @@
+class Character
+  attr_accessor :name, :level
+
+  def initialize(name: , level:)
+    self.name = name
+    self.level = level
+  end
+
+  def strength
+    @level + [*1..12].sample
+  end
+
+  def card
+    "#{name}, lvl #{level}"
+  end
+  
+end
+
+class Warrior < Character
+  
+  def initialize(name:, level:)
+    @name = name
+    @level = level
+  end
+
+end    
+
+
+class Monster < Character
+    def initialize name: , level:
+        super(name: name, level: level)
+    end
+end
+
+class BattleArena
+  
+  attr_accessor :first_character, :second_character
+  
+  def initialize(first_character, second_character)
+    self.first_character = first_character
+    self.second_character = second_character
+  end  
+
+  def levelUp(winner, loser)
+    winner.level += loser.level unless winner.is_a? Monster
```

Comment:
jeśli chodzi o używanie `unless`- jeśli bezstratnie można zmienić je na zwykłe `if`, w tym przypadku:
```
if x.is_a? Warrior
```
to tak najlepiej robić. Unless jest jednak trochę mniej naturalny i przy bardziej złożonych warunkach może sprawiać kłopoty.